payments-core/src/main/java/com/stripe/android/payments/PaymentFlowResultProcessor.kt

```diff
@@ -1,14 +1,33 @@
 package com.stripe.android.payments
 
-import com.stripe.android.PaymentIntentResult
-import com.stripe.android.SetupIntentResult
+import com.stripe.android.PaymentController
+import com.stripe.android.model.StripeIntent
 
-internal interface PaymentFlowResultProcessor {
-    suspend fun processPaymentIntent(
-        unvalidatedResult: PaymentFlowResult.Unvalidated
-    ): PaymentIntentResult
+/**
+ * Interface implemented by classes responsible for processing the result of a [PaymentController]
+ * confirm operation.
+ */
+internal interface PaymentFlowResultProcessor<StripeIntentResult> {
 
-    suspend fun processSetupIntent(
+    suspend fun processResult(
         unvalidatedResult: PaymentFlowResult.Unvalidated
-    ): SetupIntentResult
+    ): StripeIntentResult
+
+    /**
+     * It is very important to check `requiresAction()` because we can't always tell what
+     * action the customer took during payment authentication (e.g. when using Custom Tabs).
+     *
+     * We don't want to cancel if required actions have been resolved and the payment is ready
+     * for capture.
+     */
```

Comment:
Feels like these should be a inline comment inside the function, or at lest reworded a bit to begin with what this function does